### PR TITLE
fix failing pyunit_sklearn_classification_all_estimators and pyunit_s…

### PR DESCRIPTION
**Pull request by koniecsveta**
…klearn_all_estimators

### CODE REVIEW

File: h2o-algos/src/main/java/hex/rulefit/RuleFitModel.java [hunk 1]
```diff
@@ -114,7 +114,9 @@ public Frame score(Frame fr, String destination_key, Job j, boolean computeMetri
         if (_parms._weights_column != null && fr.vec(_parms._weights_column) != null)
             linearTest.add(_parms._weights_column,fr.vec(_parms._weights_column));
         
-        Frame adaptFrm = new Frame(fr);//Key.make());
+        Frame adaptFrm = new Frame(Key.make());
```
Comment (Pscheidl) [inline h2o-algos/src/main/java/hex/rulefit/RuleFitModel.java:117]: final ?

**Review by Pscheidl — APPROVED** [review]

**Review by honzasterba — APPROVED** [review]

File: h2o-algos/src/main/java/hex/rulefit/RuleFitModel.java [hunk 2]
```diff
@@ -108,13 +108,15 @@ public RuleFitModel(Key<RuleFitModel> selfKey, RuleFitParameters parms, RuleFitO
 
     @Override
     public Frame score(Frame fr, String destination_key, Job j, boolean computeMetrics, CFuncRef customMetricFunc) throws IllegalArgumentException {
```
Comment (michalkurka) [inline h2o-algos/src/main/java/hex/rulefit/RuleFitModel.java:110]: Model#score is a public method that is not expected to run in the context of an existing Scope. it is called from REST API (ModelMetricsHandler).

You can only use Scope if you are also managing it, this code doesn't appear to do that.

Same issue on line 133.

**Review by michalkurka — CHANGES_REQUESTED** [review]
Please see inline comments

**Comment by michalkurka** [comment]
Fix proposed here: https://github.com/h2oai/h2o-3/pull/5007

**Pull request closed without merging**